src/components/calcite-accordion/calcite-accordion.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Prop
+} from "@stencil/core";
+import { UP, DOWN, HOME, END } from "../../utils/keys";
+import { getElementDir } from "../../utils/dom";
+import { guid } from "../../utils/guid";
+
+@Component({
+  tag: "calcite-accordion",
+  styleUrl: "calcite-accordion.scss",
+  shadow: true
+})
+export class CalciteAccordion {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** specify the theme of dropdrown, defaults to light */
+  @Prop({ mutable: true, reflect: true }) theme: "light" | "dark" = "light";
+
+  /** specify the scale of dropdrown, defaults to m */
+  @Prop({ mutable: true, reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** specify the appearance, defaults to default */
+  @Prop({ mutable: true, reflect: true }) appearance: "default" | "minimal" =
+    "default";
+
+  /** specify the placement of the icon, defaults to end */
+  @Prop({ mutable: true, reflect: true }) iconPosition: "start" | "end" = "end";
+
+  /** specify the selection mode - defaults to multi */
+  @Prop({ mutable: true, reflect: true }) selectionMode:
+    | "multi"
+    | "single"
+    | "single-persist" = "multi";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Event() calciteAccordionItemHasChanged: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback() {
+    // validate props
+    let appearance = ["default", "minimal"];
+    if (!appearance.includes(this.appearance)) this.appearance = "default";
+
+    let iconPosition = ["start", "end"];
+    if (!iconPosition.includes(this.iconPosition)) this.iconPosition = "end";
+
+    let theme = ["light", "dark"];
+    if (!theme.includes(this.theme)) this.theme = "light";
+
+    let scale = ["s", "m", "l"];
+    if (!scale.includes(this.scale)) this.scale = "m";
+
+    let selectionMode = ["multi", "single", "single-persist"];
+    if (!selectionMode.includes(this.selectionMode))
```

Comment:
nitpick: I personally prefer to keep the braces in the if - it makes the code clearer and isn't adding that much.